### PR TITLE
development of crac creators -good practices

### PR DESCRIPTION
**Pull request by Godelaine**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private List<String> creationReport;
+
+    public CracCreationReport() {
+        creationReport = new ArrayList<>();
+    }
+
+    public void error(String errorReason) {
+        String message = String.format("[ERROR] %s", errorReason);
+        creationReport.add(message);
+        LOGGER.info(message);
```
Comment (baptsegu) [inline data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java:32]: what do you think about adapting the LOGGER level to the message ?
error -> LOGGER.error()
removed/altered/warn -> LOGGER.warn()
info -> LOGGER.info()

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public void warn(String warnReason) {
+        String message = String.format("[WARN] %s", warnReason);
+        creationReport.add(message);
+        LOGGER.info(message);
+    }
+
+    public void info(String infoReason) {
+        String message = String.format("[INFO] %s", infoReason);
```
Comment (baptsegu) [inline data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java:54]: same comment

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ */
+package com.farao_community.farao.data.crac_creator_api;
```
Comment (baptsegu) [inline data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java:4]: wrong copyright

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+public interface ElementaryCreationContext {
```
Comment (baptsegu) [inline data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java:6]: author

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ImportStatus.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.data.crac_creator_api;
+
```
Comment (baptsegu) [inline data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ImportStatus.java:8]: author

**Review by baptsegu — CHANGES_REQUESTED** [review]

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java (hunk 1 above)
Comment (pet-mit) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java]: changed

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java (hunk 2 above)
Comment (pet-mit) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreationReport.java:58]: changed

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java (hunk 3 above)
Comment (pet-mit) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java]: changed

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java (hunk 4 above)
Comment (pet-mit) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ElementaryCreationContext.java:13]: changed

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ImportStatus.java (hunk 5 above)
Comment (pet-mit) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/ImportStatus.java:9]: changed

**Review by baptsegu — APPROVED** [review]

**Pull request merged**